extensions/amp-story-player/validator-amp-story-player.protoascii

```diff
@@ -42,4 +42,40 @@ descendant_tag_list {
   tag: "A"
   tag: "SPAN"
   tag: "I-AMPHTML-SIZER"  # Only allowed when document is transformed AMP
+  tag: "IMG"
+}
+tags: {
+  html_format: AMP
+  tag_name: "IMG"
+  spec_name: "AMP-STORY-PLAYER IMG"
+  mandatory_ancestor: "AMP-STORY-PLAYER"
```

Comment:
Not mandatory parent?

Reply:
Since you are requiring data-amp-story-player-poster-img attribute as a marker, you can use that marker to indicate that this validator rule gets used for generating error messages if *and only if* it that attribute is present by adding a dispatch_key field to that attribute.

You can use:

1. `dispatch_key: NAME` which means this rule triggers iff the attribute name is present.
2. `dispatch_key: NAME_VALUE` triggers iff the attr name is present and the value is correctly set.
3. `dispatch_key: NAME_VALUE_PARENT` iff the attr name,value, and mandatory_parent all match the rule.

This is essentially a secondary key on the lookup for tagspec rules.

Reply:
Super useful to know about this! 

I missed to mention another detail earlier though: the `<img>` tag we want to include goes under the `<a>` tag under the `<amp-story-player>` tag. i.e.

```
<amp-story-player>
  <a href="story.html">
    <img amp-story-player-poster-img src="...">
  </a>
</amp-story-player>
```

So I don't think we want to set the `mandatory_parent` of the `<img>` to `amp-story-player`. I updated this to use the `dispatch_key: NAME_VALUE_PARENT_DISPATCH` and also have `mandatory_parent: "A"`. All the tests are passing as intended. Let me know if you have concerns about this.

Reply:
No strong opinion at this point, but you may then just want NAME_VALUE then.

If a user does something like:

```
<amp-story-player>
  <img amp-story-player-poster-img src="...">
</amp-story-player>
```

If you use NAME_VALUE_PARENT, because the parent is *not* `<A>`, this tagspec will not be used for producing error messages. If you use NAME_VALUE it will.

Reply:
Ohh got it. I made the changes 👍